### PR TITLE
Iter5

### PR DESCRIPTION
**Pull request by FogusB**

### CODE REVIEW

File: cmd/agent/main.go [hunk 1]
```diff
@@ -1,3 +1,100 @@
...
+	"flag"
+	"github.com/FogusB/metrics-alerts-svc/internal/collects"
+	"github.com/FogusB/metrics-alerts-svc/internal/senders"
+	"github.com/caarlos0/env/v6"
+	log "github.com/sirupsen/logrus"
+	"net/url"
+	"time"
+)
+
+type Config struct {
```
Comment (arttet) [inline cmd/agent/main.go]: SUGGESTION: it would be better if you rename it. This is not a config file.

File: cmd/agent/main.go [hunk 2]
```diff
@@ -1,3 +1,100 @@
...
+	AddressEnv        url.URL `env:"ADDRESS"`
+	AddressSrv        string
+	ReportIntervalEnv int `env:"REPORT_INTERVAL"`
+	ReportInterval    int
+	PollIntervalEnv   int `env:"POLL_INTERVAL"`
+	PollInterval      int
+}
+
+func parseFlags() (string, time.Duration, time.Duration) {
```
Comment (arttet) [inline cmd/agent/main.go]: CHANGE REQUEST: it would be better if it is in the `flags.go` file

File: cmd/agent/main.go [hunk 3]
```diff
@@ -1,3 +1,100 @@
...
+	runAddress, reportInterval, pollInterval := parseFlags()
+
+	log.Infof("Server address: %s\n", runAddress)
+	log.Infof("Report interval: %v\n", reportInterval)
+	log.Infof("Poll interval: %v\n", pollInterval)
+
+	tickerPoll := time.NewTicker(reportInterval)
+	tickerReport := time.NewTicker(pollInterval)
+
+	metrics := make(map[string]interface{})
```
Comment (arttet) [inline cmd/agent/main.go]: CHANGE REQUEST: please split your application be layers

File: cmd/agent/main.go [hunk 4]
```diff
@@ -1,3 +1,100 @@
...
+
+	go func() {
+		for {
+			select {
+			case <-tickerPoll.C:
```
Comment (arttet) [inline cmd/agent/main.go]: CHANGE REQUEST: business logic shouldn't be here.

File: cmd/server/main.go [hunk 5]
```diff
@@ -1,3 +1,44 @@
 package main
 
-func main() {}
+//go:generate go build -o=server
```
Comment (arttet) [inline cmd/server/main.go]: SUGGESTION: it would be better if it's `Makefile`

File: internal/collects/collects.go [hunk 6]
```diff
@@ -0,0 +1,44 @@
+package collects
+
+import (
+	"math/rand"
+	"runtime"
+)
+
+func CollectMetrics() map[string]interface{} {
```
Comment (arttet) [inline internal/collects/collects.go:8]: CHANGE REQUEST: You need to get them from the storage

File: internal/handlers/handlers.go [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+	if mType != storages.Gauge && mType != storages.Counter {
+		c.String(http.StatusBadRequest, "Invalid metric type")
+		log.Warning("invalid metric type")
+		return
+	}
+
+	var value storages.MetricValue
+	var err error
+	if mType == storages.Gauge {
```
Comment (arttet) [inline internal/handlers/handlers.go]: CHANGE REQUEST: please simplify this method

File: internal/routers/router.go [hunk 8]
```diff
@@ -0,0 +1,30 @@
+// Package routers содержит настройки маршрутизатора для веб-сервера.
+package routers
+
+import (
+	"github.com/FogusB/metrics-alerts-svc/internal/handlers"
```
Comment (arttet) [inline internal/routers/router.go]: CHANGE REQUEST: please sort imports

https://stackoverflow.com/questions/38704769/golang-import-grouping-by-package

**Review by arttet — CHANGES_REQUESTED** [review]
Please add the following actions in CI:
- https://about.codecov.io/blog/getting-started-with-code-coverage-for-golang/
- https://golangci-lint.run/

File: internal/collects/collects.go (hunk 6 above)
Comment (FogusB) [reply internal/collects/collects.go:8]: this is the agent package

**Review by arttet — APPROVED** [review]

**Pull request merged**